src/Microsoft.Health.Fhir.Core/Features/Search/ContinuationToken.cs

```diff
@@ -36,6 +36,19 @@ public long ResourceSurrogateId
             }
         }
 
+        public short? ResourceTypeId
+        {
+            get
+            {
+                return _tokens.Length > 1 && _tokens[^2] is long s ? (short)s : null;
```

Comment:
>is long s [](http://example.com/codeflow?start=57&length=9)

Didn't checkout branch, but from changes in code it looks like we only initilaze token with:

```
continuationToken = new ContinuationToken(new object[]
                            {
                                newContinuationType ?? 0,
                                newContinuationId ?? 0,
                            }
```

where `short? newContinuationType`

Why we cast it to long first?

Reply:
Hmm. So the JSON is deserialized as a long, which we then need to cast to a short. But you are right, the property won't work if it's an object that is created normally though the constructor.